lib/Runtime/Provisioner/Provisioner.cpp

```diff
@@ -116,17 +143,27 @@ llvm::Error Provisioner::provision(DAGListTy &networks, Module &module,
           if (backends_[i]->getBackendName() == nodeBackendName) {
             auto compiledOrErr = backends_[i]->compile(function, options);
             if (!compiledOrErr) {
+              {
```

Comment:
This code is not documented. Could you add some comments that explains the logic?